### PR TITLE
Add EdgeCore ECS platform

### PR DESCRIPTION
**Pull request by AlexLardschneider**
# Description

Added a driver for the [EdgeCore ECS](https://www.edge-core.com/productsList.php?cls=2&cls2=13) L2 switch series. These switches come preloaded with firmware made by EdgeCore, which is very similar to Cisco IOS, with only minor differences. 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

Tested this new platform on around 300 EdgeCore ECS switches, consisting of the following models:

- ECS4210-12T
- ECS4510-28T
- ECS4510-28F
- ECS4510-52P
- ECS2100-28P

This platform also seems to be working just fine on the EdgeCore AOS L3 switch series, mainly tested on model AOS5810-54X. However, I don't have access to enough devices of this series to properly test it.

I also added some pytest - tests to ensure that pattern matching is working properly.

# Checklist:

- [X] My code follows the style guidelines of this project (no GitHub actions compalints! run `make lint` before
 committing!)
- [X] I have commented my code, pydocstyle and darglint are happy, docstrings are in google docstring format, and all
 docstrings include a summary, args, returns and raises fields (even if N/A)
- [X] I have added tests that prove my fix is effective or that my feature works, if adding "functional" tests please
 note if there are any considerations for the vrnetlab setup
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scrapli_community/edgecore/ecs/_async.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Returns:
+        N/A  # noqa: DAR202
+
+    Raises:
+        N/A
+    """
+    await conn.acquire_priv(desired_priv=conn.default_desired_privilege_level)
+
+    await conn.send_command(command="terminal length 0")
+    await conn.send_command(command="terminal width 300")
```
Comment (carlmontanari) [inline scrapli_community/edgecore/ecs/_async.py:21]: guessing this is the max value? or is "0" for "unlimited" or anything an option? almost certainly thats fine just curious!

Comment (AlexLardschneider) [reply scrapli_community/edgecore/ecs/_async.py:21]: Nope, unfortunately it seems like setting the width to 0 for "unlimited" is not an option on these switches:

```
HOSTNAME-0#terminal width ?
  <80-300>  Number of character columns displayed on the terminal.
```

Comment (carlmontanari) [reply scrapli_community/edgecore/ecs/_async.py:21]: Cool, yeah thats probably no big deal anyway!

**Comment by carlmontanari** [comment]
Awesome work @AlexLardschneider thanks for contributing this!

**Pull request merged**